packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,220 @@
+
+| Prop          | Type                        | Required | Default     | Description                                                                                                                        |
+| ------------- | --------------------------- | -------- | ----------- | ---------------------------------------------------------------------------------------------------------------------------------- |
+| children      | `React.ReactNode`             | Yes       | `undefined` | Accepts multiple Checkboxes                                                                                                        |
+| label         | `string`                    | Yes       | `undefined` | The label of the group                                                                                                             |
+| labelPosition | `top, left`                 | No       | `top`       | The position of the rendered label                                                                                                 |
+| helpText    | `string`                    | No       | `undefined` | The helper text to be rendered                                                                                                     |
+| errorText    | `string`                    | No       | `undefined` | The error text to be rendered                                                                                                     |
+| defaultValue  | `string[]`                  | No       | `[]`        | The initial value of the checkbox group                                                                                            |
+| value         | `string[]`                  | No       | `[]`        | The value of the checkbox, if present will act as [controlled component](https://reactjs.org/docs/forms.html#controlled-components)                                                             |
+| onChange      | `(value: string[]) => void` | No       | `undefined` | The function to be called when any checkbox's state changes                                                                        |
+| name          | `string`                    | No       | `undefined` | The name of the checkbox group, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name) |
+| isDisabled    | `boolean`                   | No       | `false`     | Control whether the checkbox group is disabled or not.                                                                             |
+| validationState     | `undefined | 'error'`                   | No       | `undefined`     | Control whether the checkbox group is invalid or not.                                                                              |
+| necessityIndicator    | `'optional' | 'required' | undefined`                   | No       | `undefined`     | Renders optional or required indicators in the group, if set to undefined renders nothing |
```

Comment:
Checkbox has `isRequired` but CheckboxGroup has this `necessityIndicator`. Can we standardise `isRequired` for everything?

Reply:
We can't because the CheckboxGroup can also have `(optional)` and `(required)` tags in the label. 

1. If in a form all inputs are required and only few are optional we will render optional in those instead of polluting all the inputs with required tag. 
2. Similarly if in a form all inputs are optional and only few are required we will only set required on those inputs only.

Reply:
> We can't because the CheckboxGroup can also have (optional) and (required) tags in the label.

didn't understand this. So `isRequired={true}` will be same as `necessityIndicator="required"` and `isRequired={false}` will be same as `necessityIndicator="optional"` right?

> If in a form all inputs are required and only few are optional we will render optional in those instead of polluting all the inputs with required tag.
> Similarly if in a form all inputs are optional and only few are required we will only set required on those inputs only.

We can still do both of these from `isRequired` right? basically see the props of children and make decision based on that. I am guessing that is how we will do it even in `necessityIndicator` case?

Reply:
Alsoo what happens in this case-
```ts
<CheckboxGroup neccessityIndicator="optional">
  <Checkbox isRequired={true} />
  <Checkbox isRequired={false} />
  <Checkbox isRequired={false} />
</CheckboxGroup>
```

Is the first checkbox here now required or optional?

Reply:
> didn't understand this. So isRequired={true} will be same as necessityIndicator="required" and isRequired={false} will be same as necessityIndicator="optional" right?

True but technically there are 3 states. 
The last one is the absence of `necessityIndicator` will render nothing. the `necessityIndicator` is mostly a visual prop for the group. and it happens to just propagate the isRequired prop for accessibility.

Reply:
oh got it. makes sense.

What happens in this case btw https://github.com/razorpay/blade/pull/539#discussion_r924360265

Reply:
@saurabhdaware yes need to discuss on that. 

Currently if user set onChange, defaultChecked, isChecked on individual checkboxes inside the group we throw error. Maybe we also need to do this for isRequired.